dex-it/src/test/scala/com/wavesplatform/it/sync/orders/OrderDynamicFeeWithZeroFeeAccountsTestSuite.scala

```diff
@@ -0,0 +1,226 @@
+package com.wavesplatform.it.sync.orders
+
+import com.typesafe.config.{Config, ConfigFactory}
+import com.wavesplatform.dex.api.http.entities.HttpOrderStatus.Status
+import com.wavesplatform.dex.domain.account.PublicKey
+import com.wavesplatform.dex.domain.asset.Asset
+import com.wavesplatform.dex.domain.asset.Asset.{IssuedAsset, Waves}
+import com.wavesplatform.dex.domain.order.{Order, OrderType}
+import com.wavesplatform.dex.model.AcceptedOrder
+import com.wavesplatform.it.MatcherSuiteBase
+import org.scalatest.prop.TableDrivenPropertyChecks
+
+import java.util.concurrent.atomic.AtomicLong
+
+final class OrderDynamicFeeWithZeroFeeAccountsTestSuite extends MatcherSuiteBase with TableDrivenPropertyChecks {
+
+  "OrderDynamicFeeWithZeroFeeAccountsTestSuite" - {
+
+    "should match two orders with default fee for usual accounts" in test { ib =>
+      val aliceOrder = mkAliceOrder(10.waves, 1.usd, OrderType.BUY)
+      val bobOrder = mkBobOrder(10.waves, 1.usd, OrderType.SELL)
+      placeAndAwaitAtDex(aliceOrder)
+      placeAndAwaitAtDex(bobOrder, Status.Filled)
+
+      eventually {
+        wavesNode1.api.wavesBalance(alice) shouldBe ib.aliceWavesBalance + 10.waves - matcherFee
+        wavesNode1.api.wavesBalance(bob) shouldBe ib.bobWavesBalance - 10.waves - matcherFee
+        val expectedUsd = AcceptedOrder.calcAmountOfPriceAsset(10.waves, 1.usd)
+        wavesNode1.api.balance(alice, usd) shouldBe ib.aliceUsdBalance - expectedUsd
+        wavesNode1.api.balance(bob, usd) shouldBe ib.bobUsdBalance + expectedUsd
+        wavesNode1.api.wavesBalance(matcher) shouldBe ib.matcherWavesBalance + 2 * matcherFee - tradeFee
+      }
+    }
+
+    "should return reserved balance based on order for zero fee accounts" in {
+      restartWithNewZeroFeeAccounts(Set(alice))
```

Comment:
It is a some kind of luck, that this works, because we read the queue from the start (same -1 offset) after this restart and create another transactions (because of a different fee) :) 

Suggest to:
1. Create/use a different account for zero fees. 
2. Transfer to it required for tests funds.

An additional advantage is we don't have to restart the matcher.

Reply:
fxd